test/envtest/suite_test.go

```diff
@@ -135,6 +135,20 @@ func up() {
 	time.Sleep(time.Second)
 }
 
+func CleanupAfter(name string, cr client.Object) {
+	namespacedName := types.NamespacedName{Namespace: ns, Name: "intel-" + name + "-plugin"}
```

Comment:
The CRs don't seem to follow this `Name:` pattern but they are `xxxdeviceplugin-test`:

```suggestion
	namespacedName := types.NamespacedName{Namespace: ns, Name: name + "deviceplugin-test"}
```

I think with this addressed, you should be able to drop the daemonset deletion because the controller will take care of it.